src/theory/sets/cardinality_extension.cpp

```diff
@@ -132,7 +132,7 @@ void CardinalityExtension::checkCardinalityExtended(TypeNode& t)
   // get all equivalent classes of type t
   vector<Node> representatives = d_state.getSetsEqClasses(t);
 
-  if (finiteType)
+  if (finiteType && !card.isLargeFinite())
```

Comment:
Can you add a comment saying why it is safe to ignore this block for large finite types?

Reply:
Well, it is not safe as demonstrated by this example
```smt2
(set-logic ALL)
(set-option :global-declarations true)
(set-option :sets-ext true)
(declare-const A (Set (_ BitVec 117)))
(check-sat-assuming
 ((> (set.card A)  (int.pow2 117))))
```
The answer should be `unsat` but cvc5 returns `sat`, but fails to compute the model with message
```bash
(error "The model for (as set.universe (Set (_ BitVec 117))) was computed to have cardinality 166153499473114484112975882535043073. We only allow sets up to cardinality 4294967295")
```

Reply:
Ok, then we should have the same policy as for other unhandled sorts it seems, which I guess means logic exception when `finiteType` and `card.isLargeFinite()` are both true?